data_structures/src/chain.rs

```diff
@@ -1671,6 +1674,24 @@ impl TransactionsPool {
     /// assert!(!pool.vt_contains(&transaction.hash()));
     /// ```
     pub fn vt_remove(&mut self, key: &Hash) -> Option<VTTransaction> {
+        let transaction = self.vt_remove_inner(key);
+
+        if let Some(transaction) = &transaction {
+            for input in transaction.body.inputs.iter() {
+                if let Some(hashes) = self.output_pointer_map.remove(&input.output_pointer) {
+                    for hash in hashes.iter() {
+                        self.vt_remove_inner(hash);
+                        self.dr_remove_inner(hash);
+                    }
+                }
+            }
+        }
+
+        transaction
+    }
+
+    /// Value Transfer remove inner
+    fn vt_remove_inner(&mut self, key: &Hash) -> Option<VTTransaction> {
```

Comment:
Quite opaque of a function name and comment :rofl: 
```suggestion
    /// Remove a value transfer transaction from the pool without any further guarantee about removing other transactions that may try to spend the same UTXOs.
    fn vt_remove_inner(&mut self, key: &Hash) -> Option<VTTransaction> {
```